ui/StatusQ/include/StatusQ/keychain.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <QObject>
+#include <QFuture>
+
+#ifdef __OBJC__
+#include <LocalAuthentication/LAContext.h>
+#else
+class LAContext;
+#endif
+
+class Keychain : public QObject {
+    Q_OBJECT
+
+    Q_PROPERTY(QString service READ service WRITE setService NOTIFY serviceChanged)
+    Q_PROPERTY(QString reason READ reason WRITE setReason NOTIFY reasonChanged)
+    Q_PROPERTY(bool loading READ loading NOTIFY loadingChanged)
+
+public:
+    explicit Keychain(QObject *parent = nullptr);
+    ~Keychain();
+
+    enum Status {
+        StatusSuccess = 0,
+        StatusNotSupported,
+        StatusGenericError,
+        StatusUnavailable,
+        StatusCancelled,
+        StatusNotFound,
+    };
+
+    Q_ENUM(Status)
+
+    QString service() const;
+    void setService(const QString &service);
+
+    QString reason() const;
+    void setReason(const QString& reason);
+
+    bool loading() const;
+
+    Q_INVOKABLE void requestSaveCredential(const QString &account, const QString &password);
+    Q_INVOKABLE void requestDeleteCredential(const QString &account);
+    Q_INVOKABLE void requestGetCredential(const QString &account);
+    Q_INVOKABLE void cancelActiveRequest();
+
+    Status saveCredential(const QString &account, const QString &password);
+    Status deleteCredential(const QString &account);
+    Status getCredential(const QString &account, QString *out);
```

Comment:
Ideally we should keep it only in `.mm` file (in `.cpp` there is even no impl for those methods). Alternatively we could keep them private at least.

Reply:
Yeah let's make it private. And I'll add a `#ifdef Q_OS_MACOS` as well 👍